clang warning (datatypes.hpp:250:31: warning: dependent nested name specifier ...)

```
/Users/travis/build/slayoo/gdl/src/datatypes.hpp:250:31: warning: dependent nested name specifier 'Data_<Sp2>::' for friend class declaration is not supported; turning off access control for 'Data_' [-Wunsupported-friend]
  friend BaseGDL* Data_<Sp2>::Convert2( DType destTy, 
                  ~~~~~~~~~~~~^
```

--- a/src/datatypes.hpp
+++ b/src/datatypes.hpp
@@ -244,12 +244,18 @@ static	void operator delete( void *ptr);
 
   // not all compilers can handle template friend member functions
 #ifndef _MSC_VER
-#if defined( TEMPLATE_FRIEND_OK_) || (__GNUC__ >= 4)
+#  if __GNUC__ >= 4 
+#    if defined(__clang__)
+       // suppress: warning: dependent nested name specifier 'Data_<Sp2>::' for friend class declaration is not supported; turning off access control for 'Data_' [-Wunsupported-friend]
+#      pragma clang diagnostic push
+#      pragma clang diagnostic ignored "-Wunsupported-friend"
   // make all other Convert2 functions friends
   template<class Sp2>  
   friend BaseGDL* Data_<Sp2>::Convert2( DType destTy, 
 					BaseGDL::Convert2Mode);
-#else
+#      pragma clang diagnostic pop
+#    endif
+#  else
   // this explicit version does not work with GCC 4.0
   friend BaseGDL* Data_<SpDByte>::Convert2( DType destTy, BaseGDL::Convert2Mode);
   friend BaseGDL* Data_<SpDInt>::Convert2( DType destTy, BaseGDL::Convert2Mode);
@@ -266,7 +272,7 @@ static	void operator delete( void *ptr);
   friend BaseGDL* Data_<SpDComplex>::Convert2( DType destTy, BaseGDL::Convert2Mode);
   friend BaseGDL* Data_<SpDComplexDbl>::Convert2( DType destTy, BaseGDL::Convert2Mode);
 
-#endif
+#  endif
 #endif
 
   bool True();

